broker/api.py

```diff
@@ -278,59 +263,34 @@ def update(
-                        "forward_headers", []
-                    ).append("HOST")
+                if params["origin"]:
+                    origin_hostname = params["origin"]
                 else:
-                    instance.cloudfront_origin_hostname = params["origin"]
+                    origin_hostname = config.DEFAULT_CLOUDFRONT_ORIGIN
+                instance.cloudfront_origin_hostname = origin_hostname
 
             if "path" in params:
-                if params["path"] is None:
-                    instance.cloudfront_origin_path = ""
+                if params["path"]:
+                    cloudfront_origin_path = params["path"]
                 else:
-                    instance.cloudfront_origin_path = params["path"]
+                    cloudfront_origin_path = ""
+                instance.cloudfront_origin_path = cloudfront_origin_path
             if "forward_cookies" in params:
-                forward_cookies = params["forward_cookies"]
-                if forward_cookies is None:
-                    forward_cookies = "*"
-
-                forward_cookies = forward_cookies.replace(" ", "")
-                if forward_cookies == "":
-                    instance.forward_cookie_policy = (
-                        CDNServiceInstance.ForwardCookiePolicy.NONE.value
-                    )
-                    instance.forwarded_cookies = []
-                elif forward_cookies == "*":
-                    instance.forward_cookie_policy = (
-                        CDNServiceInstance.ForwardCookiePolicy.ALL.value
-                    )
-                    instance.forwarded_cookies = []
-                else:
-                    instance.forward_cookie_policy = (
-                        CDNServiceInstance.ForwardCookiePolicy.WHITELIST.value
-                    )
-                    instance.forwarded_cookies = forward_cookies.split(",")
+                forward_cookie_policy, forwarded_cookies = parse_cookie_options(params)
+                instance.forward_cookie_policy = forward_cookie_policy
+                instance.forwarded_cookies = forwarded_cookies
 
             if "forward_headers" in params:
-                forwarded_headers = params["forward_headers"]
-                if not forwarded_headers:
-                    forwarded_headers = []
-                else:
-                    forwarded_headers = forwarded_headers.replace(" ", "")
-                    forwarded_headers = forwarded_headers.split(",")
-                if (
-                    instance.cloudfront_origin_hostname
-                    == config.DEFAULT_CLOUDFRONT_ORIGIN
-                ):
-                    forwarded_headers.append("HOST")
-                instance.forwarded_headers = forwarded_headers
+                forwarded_headers = parse_header_options(params)
+            else:
+                # .copy() so sqlalchemy recognizes the field has changed
+                forwarded_headers = instance.forwarded_headers.copy()
```

Comment:
👏 - sorry this didn't cross my mind earlier when we were debugging!  I'm assuming just the [shallow copy](https://docs.python.org/3/library/copy.html#copy.copy) works here and there isn't the need of a [deep copy](https://docs.python.org/3/library/copy.html#copy.deepcopy)?

Thanks, @spgreenberg and @onelittlebecca, as you helped uncover what was going on here! :-)

Reply:
yeah, shallow copy should be fine here - nothing else can/should be messing with the list members, and all we really need to accomplish is getting sqlalchemy to see the reference change